webapi/src/test/scala/org/knora/webapi/models/FileModels.scala

```diff
@@ -0,0 +1,1131 @@
+     * @param className the class name of the resource. Defaults to "DocumentRepresentation".
+     *                  Can be a project ontology class that inherits from "DocumentRepresentation".
+     * @param shortcode the shortcode of the project to which the resource should be added.
+     *                  Defaults to "0001" (anything project).
+     * @param ontologyName the name of the ontology to be prefixed to the class name. Defaults to "knora-api".
+     *                     Should only be used, if `className` has a non-default value.
+     * @return returns a [[UploadDocumentFile]] holding a JSON-LD serialized request to create the resource as specified by the parameters.
+     */
+    def make(
+      internalFilename: String,
+      className: String = documentRepresentation,
+      shortcode: String = anythingShortcode,
+      ontologyName: String = knoraApiPrefix
+    ): UploadDocumentFile = new UploadDocumentFile(
+      UploadFileRequest
+        .make(
+          className = className,
+          internalFilename = internalFilename,
+          fileValueType = FileValueType.DocumentFileValue,
+          shortcode = shortcode,
+          ontologyName = ontologyName
+        )
+        .value
+    ) {}
+  }
+
+  /**
+   * Case class containing the JSON-LD serialization of a request to create a "knora-api:TextRepresentation".
+   *
+   * @param value the JSON-LD serialized request.
+   */
+  sealed abstract case class UploadTextFile private (value: String)
+
+  /**
+   * Companion object to the [[UploadTextFile]] case class.
+   */
+  object UploadTextFile {
+
+    /**
+     * Smart constructor for instantiating a [[UploadTextFile]].
+     *
+     * @param internalFilename the internal file name assigned by SIPI.
+     * @param className the class name of the resource. Defaults to "TextRepresentation".
+     *                  Can be a project ontology class that inherits from "TextRepresentation".
+     * @param shortcode the shortcode of the project to which the resource should be added.
+     *                  Defaults to "0001" (anything project).
+     * @param ontologyName the name of the ontology to be prefixed to the class name. Defaults to "knora-api".
+     *                     Should only be used, if `className` has a non-default value.
+     * @return returns a [[UploadTextFile]] holding a JSON-LD serialized request to create the resource as specified by the parameters.
+     */
+    def make(
+      internalFilename: String,
+      className: String = textRepresentation,
+      shortcode: String = anythingShortcode,
+      ontologyName: String = knoraApiPrefix
+    ): UploadTextFile = new UploadTextFile(
+      UploadFileRequest
+        .make(
+          className = className,
+          internalFilename = internalFilename,
+          fileValueType = FileValueType.TextFileValue,
+          shortcode = shortcode,
+          ontologyName = ontologyName
+        )
+        .value
+    ) {}
+  }
```

Comment:
I think all sealed case classes you did can be simplified to:
```
final case class UploadTextFile(
      internalFilename: String,
      className: String = textRepresentation,
      shortcode: String = anythingShortcode,
      ontologyName: String = knoraApiPrefix
)
You don't need to then create object and make method, which omits a lot of repetitions.

Reply:
Yeah something like I did it in [admin payloads](https://github.com/dasch-swiss/dsp-api/blob/main/webapi/src/main/scala/org/knora/webapi/messages/admin/responder/listsmessages/ListPayloadsADM.scala).